src/components/AddContent/AddContent.tsx

```diff
@@ -0,0 +1,479 @@
+import { RepositoryParamsResponse } from '../../services/Content/ContentApi';
+
+interface Props {
+  isLoading?: boolean;
+}
+
+const useStyles = createUseStyles({
+  description: {
+    color: global_Color_200.value,
+  },
+  removeSideBorder: {
+    '&:after': {
+      borderLeft: 'none!important',
+    },
+  },
+  toggleAllRow: {
+    composes: ['$removeSideBorder'],
+    cursor: 'pointer',
+    borderBottom: 'none!important',
+    '& td': {
+      color: global_link_Color.value + '!important',
+      padding: '12px 0 24px!important',
+    },
+    '& svg': {
+      fill: global_link_Color.value + '!important',
+      padding: '',
+    },
+  },
+  colHeader: {
+    '& td': {
+      '&:not(:last-child)': { cursor: 'pointer' },
+      padding: '12px 0!important',
+    },
+  },
+  mainContentCol: {
+    composes: ['$removeSideBorder'],
+    padding: '16px 0px 16px 36px!important',
+  },
+  toggleAction: {
+    composes: ['$removeSideBorder'],
+    '& button': {
+      padding: '8px',
+    },
+  },
+  addRepositoryButton: {
+    marginBottom: '24px',
+  },
+  saveButton: {
+    marginRight: '36px',
+  },
+  removeButton: {
+    display: 'flex!important',
+    justifyContent: 'flex-end',
+  },
+});
+
+// This adds the uniqueProperty function to the below schema validation
+Yup.addMethod(Yup.object, 'uniqueProperty', function (propertyName, message) {
+  return this.test('unique', message, function (value) {
+    if (!value || !value[propertyName]) {
+      return true;
+    }
+    if (
+      this.parent
+        .filter((v) => v !== value)
+        .some((v) => toLower(v[propertyName]) === toLower(value[propertyName]))
+    ) {
+      throw this.createError({
+        path: `${this.path}.${propertyName}`,
+      });
+    }
+
+    return true;
+  });
+});
+
+const validationSchema = Yup.array(
+  Yup.object()
+    .shape({
+      name: Yup.string().min(2, 'Too Short!').max(50, 'Too Long!').required('Required'),
+      url: Yup.string().url('Invalid URL').required('Required'),
+      gpgKey: Yup.string().optional(),
+    })
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore-next-line
+    .uniqueProperty('name', 'Names must be unique')
+    .uniqueProperty('url', 'Url\'s must be unique'),
+);
+
+const defaultValues = {
+  name: '',
+  url: '',
+  gpgKey: '',
+  gpgLoading: false,
+  expanded: true,
+};
+
+export const AddContent = ({ isLoading }: Props) => {
+  const classes = useStyles();
+  const formik = useFormik({
+    initialValues: [defaultValues],
+    validateOnBlur: false,
+    validateOnChange: false,
+    validationSchema,
+    initialTouched: [{ name: false, url: false }],
+    onSubmit: () => undefined,
+  });
+  const queryClient = useQueryClient();
+
+  const repositoryParams =
+    queryClient.getQueryData<RepositoryParamsResponse>(REPOSITORY_PARAMS_KEY);
+
+  console.log(
+    '🚀 ~ file: AddContent.tsx ~ line 142 ~ AddContent ~ repositoryParams',
+    repositoryParams,
+  );
```

Comment:
Here I access the data via queryClient.getQueryData, 
It's already been called on the main page, I now just need to use the key to get the data again.